compiler/rustc_hir_typeck/src/lib.rs

```diff
@@ -421,6 +421,18 @@ fn report_unexpected_variant_res(
             err.multipart_suggestion_verbose(descr, suggestion, Applicability::MaybeIncorrect);
             err
         }
+        Res::Def(DefKind::Variant, _) if expr.is_none() => {
+            err.span_label(span, format!("not a {expected}"));
+
+            err.span_suggestion_verbose(
+                qpath.span().shrink_to_hi().to(span.shrink_to_hi()),
+                "the struct variant's fields are being ignored",
+                " { /* field: _, field: _ */ }",
```

Comment:
Oof, I meant taking the fields themselves and emitting for each field name `name: _`. Like,

```
struct Foo { a: i32, b: i32 }
```

would result in `Foo { a: _, b: _ }`.

Reply:
Oops, I'll fix it. sorry for my misunderstanding 😭

Reply:
Additionally, I provide a diagnostic `use the struct variant pattern syntax` instead of t`he struct variant's field{s} {are} being ignored` when there are no fields.